api/v1/inferencepool_types.go

```diff
@@ -62,18 +62,29 @@ type InferencePoolSpec struct {
 	// +kubebuilder:validation:Required
 	Selector LabelSelector `json:"selector"`
 
-	// TargetPortNumber defines the port number to access the selected model server Pods.
-	// The number must be in the range 1 to 65535.
+	// TargetPorts defines the ports to access the selected model server Pods.
 	//
-	// +kubebuilder:validation:Minimum=1
-	// +kubebuilder:validation:Maximum=65535
 	// +kubebuilder:validation:Required
-	TargetPortNumber int32 `json:"targetPortNumber"`
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=1
+	// +listType=map
+	// +listMapKey=number
+	TargetPorts []Port `json:"targetPorts"`
 
 	// Extension configures an endpoint picker as an extension service.
 	ExtensionRef *Extension `json:"extensionRef,omitempty"`
 }
 
+type Port struct {
+	// Number defines the port number to access the selected model server Pods.
+	// The number must be in the range 1 to 65535.
+	//
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=65535
+	// +kubebuilder:validation:Required
+	Number int32 `json:"number"`
```

Comment:
Why are we using a concrete type for `TargetPortNumber` and an alias type for `PortNumber` when both have the same validation rules:

```go
	// +kubebuilder:validation:Minimum=1
	// +kubebuilder:validation:Maximum=65535
	// +kubebuilder:validation:Required
```

@robscott @kfswain @nirrozenbaum

Reply:
done. I use PortNumber instead of *PortNumber as it is a required field here.